webapp/app/components/pages/ActionButtons/ActionButtons.jsx

```diff
@@ -2,161 +2,215 @@ import React from 'react';
+}
+
+function isValidUser(user) {
+  return has(user, 'id') && isEmpty(user, 'id') === false;
+}
+
 export class ActionButtons extends React.Component {
   constructor(props) {
     super(props);
     
     this.state = {
+      actionType: null,
       user: {...EMPTY_USER, ...props.user},
-      modal: false,
+      isUserModalOpen: false,
       modalTitle: '',
       modalBody: {},
       modalYesLabel: '',
-      errors: cloneDeep(FORM_ERRORS)
+      errors: {}
     };
 
     this.toggle = this.toggle.bind(this);
     this.toggleAddModal = this.toggleAddModal.bind(this);
     this.toggleEditModal = this.toggleEditModal.bind(this);
     this.toggleDeleteModal = this.toggleDeleteModal.bind(this);
-    this.noUserSelected = this.noUserSelected.bind(this);
     this.updateUserState = this.updateUserState.bind(this);
     this.saveUser = this.saveUser.bind(this);
     this.cancel = this.cancel.bind(this);
-    this.isFormValid = this.isFormValid.bind(this);
+    this.validateForm = this.validateForm.bind(this);
+    this.renderUserModalBody = this.renderUserModalBody.bind(this);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.props.user.id !== nextProps.user.id) {
+  componentWillReceiveProps({ user: newUser }) {
+    const { user: currentSelectedUser } = this.props;
+
+    if (!isUserMatchById(currentSelectedUser, newUser)) {
       this.setState({
         user: {
-          ...nextProps.user
+          ...newUser
         }
       });
     }
   }
 
   toggle() {
     this.setState({
-      modal: !this.state.modal
+      isUserModalOpen: !this.state.isUserModalOpen
     });
   }
 
   toggleAddModal() {
     this.setState({
-      user: EMPTY_USER,
-      modalBody:'',
-      modalTitle: 'Add User',
-      modalYesLabel: 'Save'
+      actionType: 'add',
+      user: EMPTY_USER
     }, this.toggle);
   }
 
   toggleEditModal() {
     this.setState({
-      modalBody:'',
-      modalTitle: 'Edit User',
-      modalYesLabel: 'Save'
+      actionType: 'edit'
     }, this.toggle);
   }
 
   toggleDeleteModal() {
     this.setState({
-      modalBody: `Are you sure to delete User ${this.props.user.name}`,
-      modalTitle: 'Delete User',
-      modalYesLabel: 'Delete'
+      actionType: 'delete'
     }, this.toggle);
   }
 
-  noUserSelected() {
-    return isEmpty(this.props.user);
-
-  }
-
   updateUserState(event) {
-    const user = {
-      ...this.state.user,
-      [event.target.name]: event.target.value
-    };
-    return this.setState({user});
+    this.setState({
+      user: {
+        ...this.state.user,
+        [event.target.name]: event.target.value
+      }
+    });
   }
 
-  isFormValid() {
-    let checkErrors = cloneDeep(FORM_ERRORS);
-    if (this.state.user.name.length === 0) {
-      checkErrors.name = 'User name is required';
-    }
+  validateForm() {
+    const { user } = this.state;
+    const isValidUsername = isEmpty(user.name) === false;
+    const isValidEmail = (EMAIL_REGEXP.test(user.email) === true);
```

Comment:
You can avoid camparing to a boolean

```
const isValidUsername = !isEmpty(user.name)
const isValidEmail = EMAIL_REGEXP.test(user.email)
```

Reply:
agree, thanks will implement https://github.com/airbnb/javascript#comparison-operators--equality